tests/unit/futures_test.cc

```diff
@@ -1641,6 +1641,14 @@ SEASTAR_TEST_CASE(test_warn_on_broken_promise_with_no_future) {
     return make_ready_future<>();
 }
 
+SEASTAR_TEST_CASE(test_destroy_promise_after_state_take_value) {
+    future<> f = make_ready_future<>();
+    seastar::promise<> p;
+    f = p.get_future();
+    p.set_value();
+    return f.then([] {});
```

Comment:
Can we make the test independent on debug mode by yielding explicitly? We do have a yield() function.